File: 1013 - omaior/omaior.cpp
#include <iostream>
#include <cmath>

using namespace std;

class MaiordeTres{

    private:
        int maiordedois( int a, int b )
        {
            return((a + b + abs(a - b)) / 2) ;
        }

    public:

        int maiordetres(int a, int b, int c)
        {
            int p1, p2;
            p1 = maiordedois(a, b);
            p2 = maiordedois(p1, c);
            return p2;
        }
};

int main(void){
    int a, b, c, maior;
    MaiordeTres foo;
    cin >> a >> b >> c;
    maior = foo.maiordetres(a, b, c);
    cout << maior << " eh o maior\n";
    return 0;
}

File: 1013 - omaior/omaior.js
var input = require('fs').readFileSync('/dev/stdin', 'utf8');
var lines = input.split('\n');

function maiordedois( a, b ){
    return ((a + b + Math.abs(a - b)) / 2 );
}

function maiordetres( a, b, c ){
    p1 = maiordedois(a, b);
    return maiordedois(p1, c);
}

var a, b, c, temp, maior;
temp = lines[0].split(" ");
a = parseInt( temp[0] );
b = parseInt( temp[1] );
c = parseInt( temp[2] );
maior = maiordetres(a, b, c);
console.log("%d eh o maior", maior);
